### ISSUE TITLE
Spike: How long do activation emails take?

### ISSUE THREAD

**Issue by klemay** [hypothesis/product-backlog#988]
As mentioned in #984, users are getting stuck at the account creation process. This came up from a conversation Dan had with a potential partner, and I get up to 10 emails/month from users who say they haven't received their activation emails. 

Users expect to get the activation email instantaneously, and they can't start annotating until they click the link in the activation email. When I tested out the onboarding flow earlier today, it took about 5 minutes for my email to make its way to my inbox. 

Dan would like for us to look into:
- average turnaround times & standard deviation for activation emails
- the longest time it takes, what is the shortest time when working well, etc.
- whether Mailchimp/Mandrill is working for account creation purposes, or if we should consider other options

First step is to surface the data we need on h admin pages:
- [x] Add activation_date column to User table (#1033)
- [x] Show activation_date on user admin page (#1034) 
- [x] Don't populate last_login_date until user logs in for the first time (#1035)

**Comment by dwhly** [comment]
> I get up to 10 emails/month from users who say they haven't received their activation emails.

I wonder:  For every person that complains, how many don't?  1 to 10?  to 100?  Curious if there's a rule of thumb here.

**Comment by dwhly** [comment]
It would help if we knew when accounts were created vs when they were registered. https://github.com/hypothesis/product-backlog/issues/1027

**Comment by dwhly** [comment]
Comparison of transactional email services: https://www.ventureharbour.com/transactional-email-service-best-mandrill-vs-sendgrid-vs-mailjet/

And another: https://www.codeinwp.com/blog/best-smtp-providers/

**Comment by klemay** [comment]
Today we got a support ticket from a user who said they did not receive their activation email, but they were able to activate their account before I was able to get to the ticket. Might be an interesting data point to consider.

Support ticket: https://hypothesis.zendesk.com/agent/tickets/5352
Ticket received at: 2019-06-26 11:24am ET
User updated at: 2019-06-26 11:26am ET
Registered date on h admin page:	2019-06-26 15:16:06.636511
Mailchimp delivery time: Jun 26, 2019 11:16 am ET

Assuming the Registered field on h admin pages is UTC, we can assume the following sequence of events:
- At 11:16am, account is created 
- At 11:16am, Mandrill sends email and marks as "delivered"
- At 11:24am, user submits support ticket because they do not see activation email
- At 11:26am, user responds to their ticket because they see the activation email

Since the "account creation" and Mandrill timestamps match up, I wonder if there is a gap between when the user clicks the button to create their account and when the DB record is created + Mandrill is triggered. 

Even so, there is a 10 minute gap between when Mandrill says the email was delivered and when the user saw it.

**Comment by klemay** [comment]
A 1st step toward investigating: https://github.com/hypothesis/product-backlog/issues/1027

**Comment by klemay** [comment]
After speaking with @ajpeddakotla, we're removing this from the current sprint for a number of reasons:

- It will be much easier to get the data I need after engineers + integrations accomplish their Q3 DOs (sandboxed environments for metrics)
- LMS grading, scoped dev keys, and non-institutional PDFs with Google File picker are all much more urgent 

This is important to resolve and we shouldn't lose track of it, but it doesn't make sense to keep it in this sprint. It's in the "high priority" column of the Product board so we won't lose track of it by removing it from the sprint.

**Comment by klemay** [comment]
After analyzing a sub-set of accounts ([see slides here](https://docs.google.com/presentation/d/16o2eMrBxGNYrhimGrJM7-SuA8F11mGbnCbb0NkLFgu8/edit#slide=id.p)) we found: 

- Avg time from registration to activation: 00:35:49
- Median time from registration to activation: 00:01:00
- Max time from registration to activation: 13:29:55

We also see that Mandrill is sending confirmation emails within a minute of users creating their accounts - so the problem does not lie in the length of time it takes for Mandrill to send emails.

The current deliverability rate is 96.5%, which is not ideal.

Opening a new issue that will focus on improving account creation.
